device-bridge/rgb/animations.js

```diff
@@ -0,0 +1,68 @@
+
```

Comment:
First of all:
I would export a single function, I don't see a big need for a component

```
export default function animateGradient( from, to, duration, colorChangeCallback, completeCallback ) {
....
}
```

And I would make it use promises

```
export default function animateGradient( from, to, duration, colorChangeCallback, completeCallback = null ) {
    return new Promise( (resolve, reject ) => {
        .... async code...
       //when you want to finish:
       // So that old callback syntax works:
       if ( completeCallback ) {
            completeCallback( args );
       }
       resolve( args );
    } );
}
```